### PR TITLE
Issue #1511: Vector const copy

### PR DESCRIPTION
**Pull request by hawkfish**
Add `const` qualifiers to indicate that `VectorOperations::Copy` does not modify the source `Vector`.  A small change was needed to remove the dependence on `Vector::Normalify`, which was only using the sequence branch of the case statement.

### CODE REVIEW

File: src/common/types/chunk_collection.cpp [hunk 1]
```diff
@@ -319,38 +319,50 @@ static void QuicksortInPlace(ChunkCollection *sort_by, vector<OrderType> &desc,
...
+}
+
 // FIXME make this more efficient by not using the Value API
 // just use memcpy in the vectors
 // assert that there is no selection list
-void ChunkCollection::Reorder(idx_t order_org[]) {
-	auto order = unique_ptr<idx_t[]>(new idx_t[count]);
-	memcpy(order.get(), order_org, sizeof(idx_t) * count);
+void ChunkCollection::ReorderSlice(const slice_t slice, const idx_t *order_org) {
+	const auto slice_rows = slice.second - slice.first;
```
Comment (Mytherin) [inline src/common/types/chunk_collection.cpp:352]: Should this be part of this PR?

**Review by Mytherin — COMMENTED** [review]
Thanks for the PR! Looks fine, just one minor question:

File: src/common/types/chunk_collection.cpp (hunk 1 above)
Comment (hawkfish) [reply src/common/types/chunk_collection.cpp:352]: I couldn't figure out where it came from. Is there an easy way to remove it?

Comment (Mytherin) [reply src/common/types/chunk_collection.cpp:352]: Maybe revert the entire file and make any required changes again?

Comment (hawkfish) [reply src/common/types/chunk_collection.cpp:352]: I meant just using GitHub, but that doesn't look like an option. My house is about to lose power (utility work), so I can't get to this for a bit(!)

**Comment by Mytherin** [comment]
Thanks, looks good now!

**Pull request merged**